### PR TITLE
[perf] simplify _apply_cuda_graph_metadata for draft extend in trtllm_mla backend

### PR DESCRIPTION
**Pull request by Qiaolin-Yu**
## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #28053767452](https://github.com/sgl-project/sglang/actions/runs/28053767452)
Latest PR Test (Extra): :x: [Run #28054238156](https://github.com/sgl-project/sglang/actions/runs/28054238156)

### CODE REVIEW

File: python/sglang/srt/layers/attention/trtllm_mla_backend.py [hunk 1]
```diff
@@ -385,24 +385,13 @@ def _apply_cuda_graph_metadata(
 
         if forward_mode.is_target_verify():
             seq_lens = seq_lens[:bs] + self.num_draft_tokens
-            metadata.seq_lens_k.copy_(seq_lens.to(dtype=torch.int32))
```
Comment (kpham-sgl) [inline python/sglang/srt/layers/attention/trtllm_mla_backend.py:388]: Q: Do `TRTLLMMLADecodeMetadata` need `seq_lens_k` to be int32? `forward_batch.seq_lens` is int64 so thats why this conversion was there I think

File: python/sglang/srt/layers/attention/trtllm_mla_backend.py [hunk 2]
```diff
@@ -345,7 +345,7 @@ def _init_cuda_graph_metadata(
         if forward_mode.is_target_verify():
             metadata.seq_lens_k = torch.zeros((bs,), dtype=torch.int32, device=device)
         elif forward_mode.is_draft_extend_v2():
-            num_tokens_per_bs = num_tokens // bs
```
Comment (kpham-sgl) [inline python/sglang/srt/layers/attention/trtllm_mla_backend.py:348]: Is this correct? Other backends (like trtllm_mha) still do `num_tokens // bs` I think. Not sure if DP attention padding can affect this somehow

Comment (Qiaolin-Yu) [reply python/sglang/srt/layers/attention/trtllm_mla_backend.py:348]: https://github.com/sgl-project/sglang/pull/29077/changes#diff-394f8ec8123a9b05dc64b9733aad2eb7b768ae9e9f4d478e4be8c25f17947af6L390 in `apply_cuda_graph_metadata`, it was also `self.num_draft_tokens` before, so I think changing it here will not affect anything.

File: python/sglang/srt/layers/attention/trtllm_mla_backend.py (hunk 1 above)
Comment (Qiaolin-Yu) [reply python/sglang/srt/layers/attention/trtllm_mla_backend.py:388]: seq_lens_k is int32 buffer, so it will still do the cast when copying into the buffer, should be the same

**Review by kpham-sgl — APPROVED** [review]
